# DNA-Seq Exercises


### 1) Variant calling
	a) Run script "Exercise_exome_seq_trio.sh" step by step by copying and pasting lines of code into the terminal.  
The input files are in the ACE laptop but they can also be downloaded from 
[here exome.corpas.tar.gz](https://s3.amazonaws.com/ace-uganda/exome.corpas.tar.gz)

	b) View html file generated by snpEff and explore distribution of SNPs versus INDEL
    
    c) Look for a mutation in gene ‘MC1R’.  Is the mutation in an intron? 
    

### 2) Explore bam files in browser
* IGV: Upload bam files to IGV using File --> Upload from URL.  Enter one url at the time, then repeat to add all 3 files.  I have hosted them on cyverse (iPlant)
    * https://de.cyverse.org/anon-files/iplant/home/quinonesm/mom.chr16.realigned.bam
    * https://de.cyverse.org/anon-files/iplant/home/quinonesm/mom.chr16.realigned.bam.bai
  
    * https://de.cyverse.org/anon-files/iplant/home/quinonesm/dad.chr16.realigned.bam
    * https://de.cyverse.org/anon-files/iplant/home/quinonesm/dad.chr16.realigned.bam.bai
    
    * https://de.cyverse.org/anon-files/iplant/home/quinonesm/daughter.chr16.realigned.bam
    * https://de.cyverse.org/anon-files/iplant/home/quinonesm/daughter.chr16.realigned.bam.bai
    
    ** Note: Alternatively, upload bam files to IGV browser from your computer using File --> Load from file and selecting the file ending in ".bam".  It's important to remember that the index file (.bai) should always be in the same directory
   
### UCSC: View data in [UCSC genome browser](https://genome.ucsc.edu/) and create a custom track to view bam files

* Select Create Custom Track, then enter the following to view the daughter bam dataset
		
		track type=bam name="My BAM" bigDataUrl=https://de.cyverse.org/anon-files/iplant/home/quinonesm/daughter.chr16.realigned.bam

* Note that the dataset in this track was generated with the full dataset (not the 
* In UCSC, modify the custum track view using the options for "squish" or "full"
* (Optional) View Track Hubs of interest 
    
    
### UCSC: Annotate data using [VEP](https://www.ensembl.org/Homo_sapiens/Tools/VEP)
  


