---
title: "Summarized Experiments and Genomic Ranges"
author: "Poorani Subramanian"
date: "8/27/2018"
output: html_notebook
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, hide=TRUE)
```



## I. Create and Manipulate a SummarizeExperiment Object

```{r}
library(GenomicRanges)
library(SummarizedExperiment)
```

A Summarized Experiment object has 3 parts:
1. Assays - matrices (or data frames) - sometimes this is count data
2. The rows represent genomic regions of interest
3. The columns represent samples
4. (optional) metadata about the samples

In this small exercise, we have (fake) data for 20 genes from 9 highly talented individuals and we will create our first SummarizedExperiment object.

The data for 20 genes from 9 individuals results in a matrix:

```{r}
data <- matrix(1:180, ncol=9,  byrow=TRUE)
```

The data from the 20 genes can be represented as a GRanges object (we will learn about these later, if you'd like)

```{r}
gr_20gene <-
    GRanges(seqnames = paste0("gene", 1:20),   
    ranges = IRanges(start=1:20, width = 1000 ),  
    strand = c(rep("+", 10), rep("-", 10)),  
    score = 1:20,  
    GC = seq(1, 0, length=20))
```

The data describing the 20 individuals is stored in a data.frame

```{r}
sample_df <- data.frame( names=c("Martin", "Herve", "Dan",   
    "Marc", "Valerie", "Jim", "Nate","Paul", "Sonali"),   
    sex=c(rep("Male", 4), "Female", rep("Male", 3), "Female"))
```

### Exercises

1. Create a SummarizedExperiment from the three objects listed above.
Type `??RangedSummarizedExperiment` in the Console to see the help and parameters for the `SummarizedExperiment` function.

2. Check the dimensions of the created SummarizedExperiment (see notes from yesterday!)

3. How would you access the data matrix containing numbers?

4. How would you access the information about the genes?

5. How would you access the information about the samples?

6. Subset the SummarizedExperiment to create a new one which contains information only about the female core team.

7. Subset the SummarizedExperiment to create a new one containing only the first two genes.


## II. Count the Number of Reads Overlapping Genes

These exercises are more advanced!

### Preliminaries

This exercise requires the **GenomicFeatures** package which is a standard part of Bioconductor.  However, it may not be installed on your computer.  To check, try loading the library in your workspace:
```{r}
library(GenomicFeatures)
```

If you get an error, then you will need to install it.  This may take a few minutes
```{r}
source("https://bioconductor.org/biocLite.R")
biocLite("GenomicFeatures")
```

### Importing the data

The purpose of this exercise is to count the number of reads overlapping genes.  This is a typical first step in an RNA-seq differential expression analysis.

Annotations (gene models) can come from different sources.  We’ll use a pre-packaged source here; as well as alignments from the same Pasilla Dataset as before [PasillaBamSubset](http://bioconductor.org/packages/release/data/experiment/html/pasillaBamSubset.html)
 "Conservation of an RNA regulatory map between Drosophila and mammals" by Brooks AN, Yang L, Duff MO, Hansen KD, Park JW, Dudoit S, Brenner SE, Graveley BR, Genome Res. 2011 Feb;21(2):193-202, Epub 2010 Oct 4, PMID: 20921232. 

A later exercise retrieves  similar  information  from  a  GFF  file.  We will install the package first, then load it into memory.

```{r}
install.packages("/home/ace/ace_workshop/ace_bioconductor/TxDb.Dmelanogaster.UCSC.dm3.ensGene_3.2.2.tar.gz", repos=NULL)
library("TxDb.Dmelanogaster.UCSC.dm3.ensGene")
library(pasillaBamSubset)
```

### Exercises

1. Use `exonsBy` from the **GenomicFeatures** pkg to retrieve all exons, grouped by gene from TxDb.Dmelanogaster.UCSC.dm3.ensGene.  Subset the result to contain just those annotations relevant to chromosome 4.

2. Create a character vector of file paths, using the helper functions `untreated1_chr4` and `untreated3_chr4`.  Name
the vector with an abbreviated name derived from the file name, and use this to invoke BamFileList from **Rsamtools** pkg, creating a list of paths to files that are known to be BAM files.

We'll give the first part of the solution which is to make the vector of file paths:
```{r}
fls <- c(untreated1_chr4(), untreated3_chr4())
```


3.  Use `summarizeOverlaps` to  count  the  number  of  reads  in  each  region  of  interest of the BamFileList.   The  RNA-seq  protocol  was not  strand  aware,  so  choose `ignore.strand=TRUE`.  The  object  returned  by `summarizeOverlaps` is  an  instance of class `SummarizedExperiment`.  Explore it, and the counts that have been generate.

4. Plot  the  counts  in  the  two  samples,  perhaps  transforming  the  counts  with asinh (a  log-like  transform  that deals with 0’s better).

5.  As an additional exercise, use the rtracklayer package to retrieve and parse a GFF file containing similar annotations.

The annotations are in a GTF file on theEnsembl web site. We download them to our local disk, then usertracklayer’simportto parse the file to a GRanges instance. We subset this to contain relevant information, then split the GRanges into a GRangesList, based on the geneid column extracted from the GFF file. This could then be used in summarizeOverlaps, as above and without further change.

How would this source of annotation tie into the work flow we have just performed? (Not really an exercise - how to download and convert gff)

```{r}
library(rtracklayer)
# import gff
fl <- paste0("ftp://ftp.ensembl.org/pub/release-62/", "gtf/drosophila_melanogaster/",
"Drosophila_melanogaster.BDGP5.25.62.gtf.gz")
gffFile <- file.path(tempdir(), basename(fl))
download.file(fl, gffFile)
gff0 <- import(gffFile)
idx <- gff0$source == "protein_coding" & gff0$type == "exon" & seqnames(gff0) == "4"
gff1 <- gff0[idx]
chr4.gff <- split(gff1, mcols(gff1)$gene_id)
```


